CMakeLists.txt

```diff
@@ -0,0 +1,539 @@
+
+    add_library(ssl STATIC ${COMMON_ssl_SRCS})
+
+    target_include_directories(ssl PRIVATE include ${CMAKE_CURRENT_SOURCE_DIR})
+    target_compile_definitions(ssl PRIVATE  MD5_ASM POLY1305_ASM X25519_ASM RC4_ASM OPENSSL_CPUID_OBJ OPENSSLDIR="/usr/local/ssl" ECP_NISTZ256_ASM VPAES_ASM SHA256_ASM OPENSSL_BN_ASM_MONT5 _REENTRANT OPENSSL_IA32_SSE2 SHA1_ASM OPENSSL_PIC SHA512_ASM L_ENDIAN GHASH_ASM ENGINESDIR="/usr/local/lib/engines-1.1" KECCAK1600_ASM OPENSSL_BN_ASM_MONT OPENSSL_BN_ASM_GF2m)
+    
+    set(crypto_SRCS
+        ${COMMON_crypto_SRCS}
+        crypto/engine/eng_err.c crypto/x86_64cpuid.s crypto/whrlpool/wp-x86_64.s
+        crypto/modes/aesni-gcm-x86_64.s crypto/aes/aesni-x86_64.s crypto/camellia/cmll-x86_64.s
+        crypto/ec/x25519-x86_64.s crypto/bn/asm/x86_64-gcc.c crypto/engine/eng_init.c
+        crypto/engine/eng_fat.c crypto/bn/rsaz_exp.c crypto/ec/ecp_nistz256.c
+        crypto/sha/sha256-x86_64.s crypto/sha/sha1-mb-x86_64.s crypto/engine/tb_asnmth.c
+        crypto/ec/ecp_nistz256-x86_64.s crypto/engine/eng_ctrl.c crypto/engine/tb_eckey.c
+        crypto/engine/eng_table.c crypto/engine/eng_rdrand.c crypto/bn/x86_64-mont.s
+        crypto/engine/eng_cnf.c crypto/bf/bf_enc.c crypto/sha/sha256-mb-x86_64.s
+        crypto/aes/aesni-sha1-x86_64.s crypto/engine/tb_rand.c crypto/engine/eng_dyn.c
+        crypto/chacha/chacha-x86_64.s crypto/engine/eng_list.c crypto/sha/sha1-x86_64.s
+        crypto/engine/eng_lib.c crypto/engine/tb_cipher.c crypto/aes/aesni-mb-x86_64.s
+        crypto/des/fcrypt_b.c crypto/engine/tb_pkmeth.c crypto/rc4/rc4-md5-x86_64.s
+        crypto/rc4/rc4-x86_64.s crypto/des/des_enc.c crypto/modes/ghash-x86_64.s
+        crypto/sha/keccak1600-x86_64.s crypto/md5/md5-x86_64.s crypto/bn/x86_64-mont5.s
+        crypto/bn/rsaz-avx2.s crypto/engine/eng_all.c crypto/camellia/cmll_misc.c
+        crypto/bn/rsaz-x86_64.s crypto/aes/aes_core.c crypto/engine/tb_digest.c
+        crypto/aes/aesni-sha256-x86_64.s crypto/engine/tb_dsa.c crypto/bn/x86_64-gf2m.s
+        crypto/poly1305/poly1305-x86_64.s crypto/aes/vpaes-x86_64.s crypto/engine/tb_dh.c
+        crypto/engine/eng_pkey.c crypto/engine/eng_openssl.c crypto/sha/sha512-x86_64.s
+        crypto/engine/tb_rsa.c
+    )
+
+    add_library(crypto STATIC ${crypto_SRCS})
+
+    target_include_directories(crypto PRIVATE crypto/modes include ${CMAKE_CURRENT_SOURCE_DIR} crypto/ec/curve448/arch_32 crypto crypto/include crypto/ec/curve448)
+    target_compile_definitions(crypto PRIVATE  MD5_ASM POLY1305_ASM X25519_ASM RC4_ASM OPENSSL_CPUID_OBJ OPENSSLDIR="/usr/local/ssl" ECP_NISTZ256_ASM VPAES_ASM SHA256_ASM OPENSSL_BN_ASM_MONT5 _REENTRANT OPENSSL_IA32_SSE2 SHA1_ASM OPENSSL_PIC SHA512_ASM L_ENDIAN GHASH_ASM ENGINESDIR="/usr/local/lib/engines-1.1" KECCAK1600_ASM OPENSSL_BN_ASM_MONT OPENSSL_BN_ASM_GF2m)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-common -fPIC ")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fno-common -fPIC ")
```

Comment:
1. are there any C++ sources to need the CXX flags?
2. does cmake fork/restore these flags when processing submodules or will they propagate to other targets?

Reply:
There are not, but they won't hurt in case we need make changes to OpenSSL, since CMAKE_CXX_FLAGS won't be seen in the outer scope unless PARENT_SCOPE argument is provided

Reply:
well, if we do make changes, it will definitely be in C :)
but as long as flags don't leak, I'm fine with it